interpreter/mesh/src/main/scala/io/buoyant/interpreter/mesh/Client.scala

```diff
@@ -49,7 +51,29 @@ object Client {
     delegator: mesh.Delegator,
     backoffs: scala.Stream[Duration],
     timer: Timer
-  ) extends NameInterpreter with Delegator {
+  ) extends NameInterpreter with Delegator with Admin.WithHandlers {
+
+    private case class InstrumentedBind(
+      act: InstrumentedActivity[NameTree[Name.Bound]],
+      stream: StreamState[mesh.BindReq, mesh.BoundTreeRsp]
+    )
+    private case class InstrumentedResolve(
+      act: InstrumentedVar[Addr],
+      stream: StreamState[mesh.ReplicasReq, mesh.Replicas]
+    )
+    private case class InstrumentedDtab(
+      act: InstrumentedActivity[Dtab],
+      stream: StreamState[mesh.DtabReq, mesh.DtabRsp]
+    )
+
+    private[this] val bindCacheMu = new {}
+    @volatile private[this] var bindCache = Map.empty[(Dtab, Path), InstrumentedBind]
```

Comment:
It would be good, as a followup, to investigate which caching techniques are friendliest to the jvm (i.e. Map, mutable.Map, CHM, etc)